packages/schematics/src/command-line/visualization.ts

```diff
@@ -0,0 +1,316 @@
+  return nodeProps[getVariant(criticalPath, projectNode.name)];
+}
+
+export function getEdgeProps(
+  config: GraphvizOptionNodeEdge,
+  depType: DependencyType,
+  child: string,
+  criticalPath: CriticalPathMap
+) {
+  const edgeProps = config[depType];
+  return edgeProps[getVariant(criticalPath, child)];
+}
+
+export function createGraphviz(
+  config: GraphvizConfig,
+  depTree: Dep,
+  projectMap: ProjectMap,
+  criticalPath: CriticalPathMap
+) {
+  var g = graphviz.digraph('G');
+  config.graph.forEach(({ attr, value }) => g.set(attr, value));
+
+  Object.keys(depTree)
+    .sort()
+    .forEach(key => {
+      // sorting helps with testing
+      const projectNode = projectMap[key];
+      const dependencies = depTree[key];
+
+      g.addNode(key, getNodeProps(config.nodes, projectNode, criticalPath));
+
+      if (dependencies.length > 0) {
+        dependencies.forEach((dep: Dependency, i: number) => {
+          g.addNode(
+            dep.projectName,
+            getNodeProps(config.nodes, projectNode, criticalPath)
+          ); // child node
+
+          g.addEdge(
+            key,
+            dep.projectName,
+            getEdgeProps(config.edges, dep.type, dep.projectName, criticalPath)
+          );
+        });
+      }
+    });
+
+  return g;
+}
+
+export enum OutputType {
+  'json' = 'json',
+  'html' = 'html',
+  'dot' = 'dot'
+}
+
+function handleOutput(
+  outputType: OutputType,
+  outputTypes: OutputType[],
+  output: string,
+  noOpen: boolean,
+  filename?: string,
+  app?: string
+) {
+  if (outputTypes.length !== 0 && outputTypes.indexOf(outputType) === -1) {
+    return;
+  }
+
+  if (outputTypes.length !== 0 && !filename) {
+    return console.log(output);
+  }
+
+  const path = `.vis/${filename || 'visualization.' + outputType}`;
+  writeToFile(path, output);
+
+  if (outputTypes.length === 0 || noOpen) {
+    return;
+  }
+
+  opn(path, {
+    wait: false,
+    ...(app ? { app } : undefined)
+  });
+}
+
+function applyHTMLTemplate(visData: string) {
+  return `<!DOCTYPE html>
+  <html>
+    <head><title></title></head>
+    <body>${visData}</body>
+  </html>
+  `;
+}
+
+function renderGraph(
+  affected?: string,
+  outputTypes: OutputType[] = [],
+  noOpen: boolean = false,
+  filename?: string,
+  app?: string
+) {
+  const config = readCliConfig();
+  const npmScope = config.project.npmScope;
+  const projects: ProjectNode[] = getProjectNodes(config);
+  const projectMap: ProjectMap = mapProjectNodes(projects);
+
+  // fetch all apps and libs
+  const depTree = dependencies(npmScope, projects, f =>
+    readFileSync(`${appRoot.path}/${f}`, 'utf-8')
+  );
+
+  // fetch apps and libs on the critical path
+  const criticalPath = allAffected(
+    npmScope,
+    projects,
+    f => readFileSync(`${appRoot.path}/${f}`, 'utf-8'),
+    affected ? affected.split(',') : []
+  );
+
+  const json = {
+    depTree,
+    criticalPath
+  };
+
+  const graphviz = createGraphviz(
```

Comment:
I'd change createGraphviz to accept:

config, deps, projects, and affectedProjects and calculate the needed maps in it.